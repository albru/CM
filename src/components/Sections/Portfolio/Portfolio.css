.Portfolio {
    width: 100%;
    position: relative;
    text-align: center;
    margin-bottom: 40px;
};