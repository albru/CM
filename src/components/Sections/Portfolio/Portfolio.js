import React from 'react';
import { connect } from 'react-redux';

import * as actionTypes from '../../../store/actions/actions';
import PortfolioItems from './Portfolioitems/PortfolioItems';
import BreadCrumbs from '../../Navigation/BreadCrumbs/BreadCrumbs';
import Button from  '../../UI/Button/Button';
import classes from './Portfolio.css';

const portfolio = props => {
    console.log(props)
    const showMoreButton = (
        props.maximumItems === false ?
         <Button clicked={props.showNewPortfolioItem} 
            btnType="MainButton">Показать еще</Button> :
         null
    );

    const crumbs = [
        {
            crumbName:'Главная',
            crumbHref: '/',
            linkType: 'enabled'},
        {
            crumbName:'Портфолио',
            crumbHref: '/portfolio',
            linkType: 'disabled'
        }
    ]
    return (
        <section className={classes.Portfolio}>
            <BreadCrumbs crumbs={crumbs} />
            <h1>Работы</h1>
            <PortfolioItems />
            {showMoreButton}
        </section>
    )
};

const mapStateToProps = state => {
    return {
        maximumItems: state.portfolio.maximumItems
    }
}

const mapDispatchToProps = dispatch => {
    return {
        showNewPortfolioItem: () => dispatch({type: actionTypes.SHOW_MORE_PORTFOLIO_ITEMS})
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(portfolio);