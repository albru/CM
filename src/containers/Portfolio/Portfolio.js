import React from 'react';
import Aux from '../../hoc/_Aux/_Aux';
import Portfolio from '../../components/Sections/Portfolio/Portfolio';

const portfolio = props => {
    return (
        <Aux>
            <Portfolio />
        </Aux>
    )
};

export default portfolio;