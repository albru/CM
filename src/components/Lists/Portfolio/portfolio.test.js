import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Portfolio from './Portfolioitems/PortfolioItems';

configure({ adapter: new Adapter() })

describe('PortfolioItems', () => {
    it('should render correctly', () => {
        const props = {
            dataArr: []
        }
        const output = shallow(
            <Portfolio {...props}/>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});