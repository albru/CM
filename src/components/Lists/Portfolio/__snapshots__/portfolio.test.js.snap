// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioItems should render correctly 1`] = `<ul />`;
